documentation/error-handling.md

```diff
@@ -0,0 +1,113 @@
+# Error handling
+
+Because UI extension scripts run in non-browser environments (Web Worker in web, JavascriptCore on Mobile), it is recommended to manually set up error listeners and error reporting. However, it is possible to use external libraries for error reporting.
+
+## Error handling in UI extensions
+
+UI extensions can throw two types of error: lifecycle errors and non-lifecycle errors.
+
+### Lifecycle error
+
+Most errors will fall into this type. These errors can be thrown when:
+
+- a UI extension script is loaded or imported
+- the UI extension script is run
+- during any callback interaction with UI extension components or APIs
+
+To catch these errors, add an event listener for `unhandledrejection`.
+
+  ```javascript
+  self.onunhandledrejection = (error) => {
+    console.warn('onunhandledrejection', error);
+  };
+  // or
+  self.addEventListener('unhandledrejection', (error) => {
+    console.warn('event unhandledrejection', error);
+  });
+  ```
+
+### Non-lifecycle error
+
+These errors can be thrown when the script encounters an exception while running a `setTimeout` callback. To catch these errors, add an event listener for `error`.
+
+  ```javascript
+  self.onerror = (error) => {
+    console.warn('onerror', error);
+  };
+  // or
+  self.addEventListener('error', (error) => {
```

Comment:
Can we pick one of the formats and recommend only that one? I think we should just do the more modern addEventListener() examples.